### PR TITLE
Feature/update unit tests for assets

### LINKED ISSUE #457: Update unit tests in app/services/assets/assets_test.go

### PR DESCRIPTION
**Pull request by nnedkov-limechain**
**Detailed description**:
Updating tests for service.Assets and adding token clients interface to be able to mock it. Initializing all token clients on start so to be reused everywhere.

**Which issue(s) this PR fixes**:
Fixes #457 

**Special notes for your reviewer**:

**Checklist**
- [ ] Documentation added
- [x] Tests updated

### CODE REVIEW

**Comment by failfmi** [comment]
E2E fails with null pointer exception

File: e2e/e2e_test.go [hunk 1]
```diff
@@ -578,7 +578,7 @@ func Test_EVM_Native_to_EVM_Token(t *testing.T) {
 	chainId := uint64(80001) // represents Polygon Mumbai Testnet (e2e config must have configuration for that particular network)
 	evm := setupEnv.Clients.EVM[chainId]
 	now = time.Now()
-	targetChainID := uint64(43113) // represents Avalanche Fuji Testnet (e2e config must have configuration for that particular network)
+	targetChainID := uint64(3) // represents Avalanche Fuji Testnet (e2e config must have configuration for that particular network)
```
Comment (failfmi) [inline e2e/e2e_test.go:581]: Can we revert it to 43113

File: e2e/e2e_test.go [hunk 2]
```diff
@@ -653,8 +653,8 @@ func Test_EVM_Wrapped_to_EVM_Token(t *testing.T) {
 	// Step 1 - Initialize setup, smart contracts, etc.
 	setupEnv := setup.Load()
 
-	chainId := uint64(80001)     // represents Polygon Mumbai Testnet (e2e config must have configuration for that particular network)
-	sourceChain := uint64(43113) // represents Avalanche Fuji Testnet (e2e config must have configuration for that particular network)
```
Comment (failfmi) [inline e2e/e2e_test.go:657]: Can we revert it to 43113

File: bootstrap/watchers.go [hunk 3]
```diff
@@ -75,7 +75,9 @@ func createPrometheusWatcher(
 	configuration config.Config,
 	prometheusService service.Prometheus,
 	EVMClients map[uint64]client.EVM,
+	EVMTokenClients map[uint64]map[string]client.EVMToken,
```
Comment (vectorman1) [inline bootstrap/watchers.go:78]: evm should be lowercase in both :)

File: e2e/setup/config.go [hunk 4]
```diff
@@ -186,15 +186,19 @@ type clients struct {
 	Distributor     service.Distributor
 }
 
-func getRouterAndEVMClientsFromEVMUtils(evmUtils map[uint64]EVMUtils) (routerClients map[uint64]*router.Router, evmClients map[uint64]client.EVM) {
+func getRouterAndEVMTokenClientsFromEVMUtils(evmUtils map[uint64]EVMUtils) (routerClients map[uint64]*router.Router, evmTokenClients map[uint64]map[string]client.EVMToken) {
```
Comment (vectorman1) [inline e2e/setup/config.go:189]: Omit `get`

**Review by vectorman1 — CHANGES_REQUESTED** [review]

File: test/mocks/evm-token-client/evm_token_client_mock.go [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+package evm_token_client
+
+import (
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/stretchr/testify/mock"
+	"math/big"
+)
+
+type MockEVMTokenClient struct {
```
Comment (vectorman1) [inline test/mocks/evm-token-client/evm_token_client_mock.go:26]: This should be unexported w/ an exported ctor

File: test/mocks/setup.go [hunk 6]
```diff
@@ -49,6 +50,7 @@ var MEVMCoreClient *evm_client.MockEVMCoreClient
 var MHTTPClient *http_client.MockHttpClient
 var MReadOnlyService *service.MockReadOnlyService
 var MEVMClient *evm_client.MockEVMClient
+var MEVMTokenClient *evm_token_client.MockEVMTokenClient
```
Comment (vectorman1) [inline test/mocks/setup.go:53]: These here can all be interfaces

and later given concrete mock implementations in the setup below

File: app/services/assets/assets.go [hunk 7]
```diff
@@ -67,35 +66,35 @@ func (a *Service) WrappedToNative(wrappedAsset string, wrappedChainId uint64) *a
...
 }
 
 func (a *Service) IsNative(networkId uint64, asset string) bool {
 	_, isNative := a.nativeToWrapped[networkId][asset]
 	return isNative
 }
 
-func (a *Service) GetOppositeAsset(sourceChainId uint64, targetChainId uint64, asset string) string {
-	nativeAssetForTargetChain := a.WrappedToNative(asset, sourceChainId)
+func (a *Service) GetOppositeAsset(chainOne uint64, chainTwo uint64, asset string) string {
```
[the remark this answers is not in the document]
Comment (nnedkov-limechain) [reply app/services/assets/assets.go:82]: Will remove the Get but the names of the parameters are correct like that as the source and target in the name is misleading as in this case we don't have direction. We just need to get the opposite asset for the specified chains.

File: bootstrap/watchers.go (hunk 3 above)
Comment (nnedkov-limechain) [reply bootstrap/watchers.go:78]: ops yep

File: test/mocks/evm-token-client/evm_token_client_mock.go (hunk 5 above)
Comment (nnedkov-limechain) [reply test/mocks/evm-token-client/evm_token_client_mock.go:26]: Please open an issue for refactoring all of those types.

File: test/mocks/setup.go (hunk 6 above)
Comment (nnedkov-limechain) [reply test/mocks/setup.go:53]: Please open an issue for refactoring all of those types.

File: app/services/assets/assets_test.go [hunk 8]
```diff
@@ -17,65 +17,213 @@
...
+	setup()
+
+	actual := serviceInstance.FungibleNetworkAssets()
+	expected := testConstants.FungibleNetworkAssets
+
+	assert.NotNil(t, actual)
+	assert.Equal(t, expected, actual)
+}
+
+func Test_GetNativeToWrappedAssets(t *testing.T) {
```
Comment (svetlio8) [inline app/services/assets/assets_test.go:162]: `Test_NativeToWrappedAssets ` instead `Test_GetNativeToWrappedAssets `?

File: app/services/assets/assets_test.go [hunk 9]
```diff
@@ -17,65 +17,213 @@
...
+	setup()
+
+	actual := serviceInstance.FungibleNativeAsset(constants.HederaNetworkId, constants.Hbar)
+	expected := testConstants.Network0FungibleNativeAsset
+
+	assert.NotNil(t, actual)
+	assert.Equal(t, expected, actual)
+}
+
+func Test_GetFungibleAssetInfo(t *testing.T) {
```
Comment (svetlio8) [inline app/services/assets/assets_test.go:202]: `Test_FungibleAssetInfo` instead `Test_GetFungibleAssetInfo` ?

**Review by svetlio8 — APPROVED** [review]

**Review by vectorman1 — APPROVED** [review]

**Pull request merged**